h3i/src/recordreplay/qlog.rs

```diff
@@ -599,14 +599,13 @@ mod tests {
             importance: qlog::events::EventImportance::Core,
             name: H3I_WAIT.to_string(),
             data: serde_json::to_value(WaitType::WaitDuration(
-                Duration::from_millis(0),
+                Duration::from_millis(12345),
             ))
             .unwrap(),
         };
         let serialized = serde_json::to_string(&ev);
 
-        let expected =
-            r#"{"time":123.0,"name":"h3i:wait","data":{"secs":0,"nanos":0}}"#;
+        let expected = r#"{"time":123.0,"name":"h3i:wait","data":12345}"#;
```

Comment:
In qlog, the `data` field is always a structure. So this should be something like 

`r#"{"time":123.0,"name":"h3i:wait","data":{"duration":12345}}"#;`

i.e. the `WaitDuration` enum variant should either contain a structure, or the ser/der handles tagging it

Reply:
Ok, fixed.